chain/actions/src/payload.rs

```diff
@@ -36,43 +36,48 @@ pub enum Operation {
 pub trait PayloadGenerator<T: Into<TypedTransaction>> {
     /// Create a ERC20 approve transaction payload. Pre-requisite to open payment channels.
     /// The `spender` address is typically the HOPR Channels contract address.
-    fn approve(&self, spender: Address, amount: Balance) -> Result<T>;
+    fn approve(&self, spender: &Address, amount: &Balance) -> Result<T>;
 
     /// Create a ERC20 transfer transaction payload
-    fn transfer(&self, destination: Address, amount: Balance) -> Result<T>;
+    fn transfer(&self, destination: &Address, amount: &Balance) -> Result<T>;
 
     /// Creates the transaction payload to announce a node on-chain.
-    fn announce(&self, announcement: AnnouncementData) -> Result<T>;
+    fn announce(&self, announcement: &AnnouncementData) -> Result<T>;
 
     /// Creates the transaction payload to open a payment channel
-    fn fund_channel(&self, dest: Address, amount: Balance) -> Result<T>;
+    fn fund_channel(&self, dest: &Address, amount: &Balance) -> Result<T>;
 
     /// Creates the transaction payload to immediately close an incoming payment channel
-    fn close_incoming_channel(&self, source: Address) -> Result<T>;
+    fn close_incoming_channel(&self, source: &Address) -> Result<T>;
 
     /// Creates the transaction payload that initiates the closure of a payment channel.
     /// Once the notice period is due, the funds can be withdrawn using a
     /// finalizeChannelClosure transaction.
-    fn initiate_outgoing_channel_closure(&self, destination: Address) -> Result<T>;
+    fn initiate_outgoing_channel_closure(&self, destination: &Address) -> Result<T>;
 
     /// Creates a transaction payload that withdraws funds from
     /// an outgoing payment channel. This will succeed once the closure
     /// notice period is due.
-    fn finalize_outgoing_channel_closure(&self, destination: Address) -> Result<T>;
+    fn finalize_outgoing_channel_closure(&self, destination: &Address) -> Result<T>;
 
     /// Used to create the payload to claim incentives for relaying a mixnet packet.
-    fn redeem_ticket(&self, acked_ticket: AcknowledgedTicket, domain_separator: Hash) -> Result<T>;
+    fn redeem_ticket(
+        &self,
+        chain_key: &ChainKeypair,
+        acked_ticket: &AcknowledgedTicket,
+        domain_separator: &Hash,
+    ) -> Result<T>;
 
     /// Creates a transaction payload to register a Safe instance which is used
     /// to manage the node's funds
-    fn register_safe_by_node(&self, safe_addr: Address) -> Result<T>;
+    fn register_safe_by_node(&self, safe_addr: &Address) -> Result<T>;
```

Comment:
`Address` and `Balance` are small types, passing them by reference has almost no performance gain. They are both `Copy`.

Reply:
changed